framework/src/play/libs/ws/WSAsync.java

```diff
@@ -663,6 +667,43 @@ public InputStream getStream() {
             }
         }
 
+        @Override
+        public Document getXml(String encoding) {
+            try {
+                checkConsumed();
+                return super.getXml(encoding);
+            } finally {
+                consumed = true;
```

Comment:
the idea is to allow getString() and getXml() calls only once, on exceptions it's probable that the stream is already partially consumed.